runners/spark/src/main/java/org/apache/beam/runners/spark/translation/EvaluationContext.java

```diff
@@ -231,4 +252,8 @@ private String storageLevel() {
     return runtime.getPipelineOptions().as(SparkPipelineOptions.class).getStorageLevel();
   }
 
+  public Map<PCollection, Long> getCacheCandidates() {
```

Comment:
pull this up to where all `public` methods are, and add Javadoc. 
Also, could you do me a favour and remove:
```
/**
   * Retrieves an iterable of results associated with the PCollection passed in.
   *
   * @param pcollection Collection we wish to translate.
   * @param <T>         Type of elements contained in collection.
   * @return Natively types result associated with collection.
   */
  <T> Iterable<T> get(PCollection<T> pcollection) {
    Iterable<WindowedValue<T>> windowedValues = getWindowedValues(pcollection);
    return Iterables.transform(windowedValues, WindowingHelpers.<T>unwindowValueFunction());
  }
```
it's unused and I always forget to remove it.